Sources/SPTDataLoader.m

```diff
@@ -130,6 +133,26 @@ - (void)removeRequest:(SPTDataLoaderRequest *)request
     return cancellationToken;
 }
 
+- (nullable id<SPTDataLoaderCancellationToken>)performRequest:(SPTDataLoaderRequest *)request
+                                               withCompletion:(void (^)(SPTDataLoaderResult *result))completion
+{
+    SPTDataLoaderRequest *copiedRequest = [request copy];
+    id<SPTDataLoaderCancellationToken> cancellationToken = [self.cancellationTokenFactory createCancellationTokenWithDelegate:self
+                                                                                                                 cancelObject:copiedRequest];
+    copiedRequest.cancellationToken = cancellationToken;
+    @synchronized(self.cancellationTokens) {
+        [self.cancellationTokens addObject:cancellationToken];
+    }
+    @synchronized(self.requests) {
+        [self.requests addObject:copiedRequest];
+    }
+    @synchronized(self.requestHandlers) {
+        [self.requestHandlers setObject:completion forKey:request];
```

Comment:
Shouldn't this use `copiedRequest`?

Reply:
apparently not.

Reply:
@dmakarov can you elaborate why `copiedRequest` is not needed here?

Reply:
Initially I used the copiedRequest as a key and it didn't work.  Using the actual request is correct.

Reply:
I guess because as key it uses the address of the object so you need to original one to retrieve it later.